File: Application/App/src/app/models/matchDetails/objectives-dto.ts
import { ObjectiveDto } from "./objective-dto"

export class ObjectivesDto {
    baron: ObjectiveDto = new ObjectiveDto({});
    champion: ObjectiveDto = new ObjectiveDto({});
    dragon: ObjectiveDto = new ObjectiveDto({});
    inhibitor: ObjectiveDto = new ObjectiveDto({});
    riftHerald: ObjectiveDto = new ObjectiveDto({});
    tower: ObjectiveDto = new ObjectiveDto({});

    public constructor(data : Partial<ObjectivesDto>){
        Object.assign(this,data);
    }
}

File: Backend/Backend/Models/Matches/ObjectivesDto.cs
namespace Backend.Models.Matches
{
    public class ObjectivesDto
    {
        public ObjectivesDto(ObjectiveDto baron, ObjectiveDto champion, ObjectiveDto dragon,
            ObjectiveDto inhibitor, ObjectiveDto riftHerald, ObjectiveDto tower)
        {
            Baron = baron;
            Champion = champion;
            Dragon = dragon;
            Inhibitor = inhibitor;
            RiftHerald = riftHerald;
            Tower = tower;
        }

        public ObjectiveDto Baron { get; set; }
        public ObjectiveDto Champion { get; set; }
        public ObjectiveDto Dragon { get; set; }
        public ObjectiveDto Inhibitor { get; set; }
        public ObjectiveDto RiftHerald { get; set; }
        public ObjectiveDto Tower { get; set; }
    }
}
